### ISSUE TITLE
Create research projects

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#1100]
**Is your feature request related to a problem? Please describe.**
- Currently in mars-sim, many scientific studies are being carried out.
- One possibility is to make use of these studies to do applied research 

**Describe the solution you desire to see**
- Create the concept of the *project collaboration*
- In a typical *project collaboration*, it involves the followings : 
1. Assemble a multi-disciplinary team of researchers and engineers
2. Create a blueprint and define a design prototype
3. Choose a site (an existing settlement, a remote site, etc.)
4. Set up a lab in a test site
5. Design and test the prototype product
6. Sell the finished design or build factory for mass production of final product
7. Improve the design and get ready for producing the 2nd generation of this product

**Describe the rationale**
- The general goals for these research projects should be within the goals of the sponsors as stated in `governance.xml`

**Specifications  (please complete the following information):**
 - pre 3.7.0  build 8798

**Additional context**
- In future, we may consider how owning these blueprints and/or design prototypes would raises the economic prospect, national prestige and industrial capability of a settlement

**Comment by Urwumpe** [comment]
I think, we should separate the scientific question from the project to research it. This way, we could easier plan, how a research lab has to be equipped and how long it will be used in a certain configuration. And it would be easier to tell, if a settlement needs more labs or workshops in advance. 

Also, we could then plan travel and personal transfers, like one scientist moving into a new settlement, because there is a competence cluster for a certain field or type of research.

**Comment by mokun** [comment]
> I think, we should separate the scientific question from the project to research it.

@Urwumpe @bevans2000 

Yes. A scientific inquiry doesn't always immediately produce an application.

The idea of Research and Development (R&D) is NOT always making sense.

*Basic* research serves to answer scientific questions.

*Applied* research gears toward developing or improving a technology.

Some of those sponsors' mission agendas are *basic* research only with no clear application in sight.

But some mission agendas suggest applied research can be done and a design prototype can be developed.

So, may be we leave our existing Scientific Study untouched as much as possible ?!

But we will go ahead to model how some of these researchers will get hired or consulted to do *Applied* Research that will have potential real-world applications.

In general, we will further refine and modify how those sponsors' mission agendas are worded in near future as we begin to code and define exactly what each application should look like, how to quantify it, etc. in mars-sim.

**Comment by bevans2000** [comment]
I think we shouldn't change Scientific Study because have scheduled this to be converted to the generic Project approach.

**Comment by mokun** [comment]
> shouldn't change Scientific Study because have scheduled this to be converted to the generic Project approach.

Ok. Then I'll wait for your conversion to get done first. 

Q: Is it your next task or next next ?

``

**Comment by mokun** [comment]
But @bevans2000  how do you like this R&D approach so far ?

I recall that @Urwumpe mentioned a few times that he's very interested in pushing forward the concept of research project as a way to go beyond what we currently have in doing just scientific study ...

**Comment by Urwumpe** [comment]
Exactly, I was looking for ways to make R&D more rewarding and have more lasting effects on the simulation outcome. Right now, the research is just for a score and more like basic research. For achieving this, there are many options though: 

- Basic Research for gaining sponsor favor and support
- Basic Research for improving abstract underlying technologies
- Basic research to improve greenhouse yields
- Basic Research to develop new medical cures
- Applied Research as way of product development (rovers, robots, science tools, EVA suits, ? furniture ?)
- Applied Research as way to extend the base
- Applied Research to develop training materials for the settlers
- Applied Research to find new routes to sites
- ...

**Comment by bevans2000** [comment]
I like the idea of research project; eventually I think it will replace the current Scientific Study. We need to tie these to the objectives of the Reporting Authority.
We can then define the projects and benefits in an external XML file. The benefits that are delivered can be generic modifiers that are captured using the existing Settlement Preferences approach. This will make it extendable via configurations.
So we define a set of benefit modifiers that are used inside the various coded logic calculations inside the simulation. For example "greenhouse yield" would be used by the Crop class. The driving modifier could be updated via many paths, e.g. Research Project, Reporting Authority objectives, or a Commander adjusting it.
My goal with #963 was to build a generic (& single) approach to Settlement modifiers.

**Comment by bevans2000** [comment]
> > shouldn't change Scientific Study because have scheduled this to be converted to the generic Project approach.
> 
> Ok. Then I'll wait for your conversion to get done first.
> 
> Q: Is it your next task or next next ?
> 
> ``
I am super busy at work but I'm focusing on the Issues assigned to 3.7.0. First is #1099 and then #1084 so we have a baselined blueprint. The latter will take time over a longer period; more of a background activity.

**Comment by mokun** [comment]
> shouldn't change Scientific Study because have scheduled this to be converted to the generic Project approach.

By the way, when you convert the Scientific Study to the Generic Project approach, would you also consider how we may make `Member` class compatible with, say, the Colonist class (currently it has nothing inside) ?　

This way, researchers from lunar colonists or those from any colonies elsewhere in the solar system may opt in for scientific research.

I've been finding ways to simplify or decouple things in `Person` so that I can come up with a simple Colonist class that would incorporate the basics of a `Person` that can serve as a Researcher. But that's a lot of changes and it may not be successful at the end.

@Urwumpe @bevans2000 
How do you see we can get it done the quickest and easiest to ?  

This is a part of https://github.com/mars-sim/mars-sim/issues/714.
